java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcToArrowVectorIteratorTest.java

```diff
@@ -99,15 +104,25 @@ public void testJdbcToArrowValues() throws SQLException, IOException {
 
   @Test
   public void testVectorSchemaRootReuse() throws SQLException, IOException {
-    String[] expectedColValues = {
+    String[] expectedIntColValues = {
         "[101, 102, 103]",
         "[104, null, null]",
         "[107, 108, 109]",
         "[110]"
     };
+    String[] expectedArrayColValues = {
```

Comment:
Added this test case for arrays specifically to test the fix to `ArrayConsumer` when resetting the delegate. Without the reset you end up reading null values.